### PR TITLE
Portlet-PO for Build Statistics and simple tests

### LINKED ISSUE #2: Build statistics Portlet
Das Standard-Portlet "Build statistics" soll getestet werden.

### PR DESCRIPTION
**Pull request by darenegade**
closes #2

### CODE REVIEW

**Comment by darenegade** [comment]
@uhafner Bitte einmal drüber schauen. Das ist der erste PR mit Tests für ein Portlet

File: src/main/java/org/jenkinsci/test/acceptance/plugins/dashboard_view/BuildStatisticsPortlet.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+/**
+ * The basic Build statistics portlet shipped with the dashboard view plugin.
+ *
+ * @author Rene Zarwel
+ */
+@Describable("Build statistics")
+public class BuildStatisticsPortlet extends AbstractDashboardViewPortlet {
+
+  public enum Jobtype{
```
Comment (uhafner) [inline src/main/java/org/jenkinsci/test/acceptance/plugins/dashboard_view/BuildStatisticsPortlet.java]: [Coding style:](https://github.com/uhafner/codingstyle/blob/master/doc/Formatierung.md) whitespace before a brace.

File: src/main/java/org/jenkinsci/test/acceptance/plugins/dashboard_view/BuildStatisticsPortlet.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    FAILED(2), UNSTABLE(3), SUCCESS(4), PENDING(5),
+    DISABLED(6), ABORTED(7), NOT_BUILT(8), TOTAL(9);
+
+    private final int row;
+    Jobtype(int r){
+      row =r;
```
Comment (uhafner) [inline src/main/java/org/jenkinsci/test/acceptance/plugins/dashboard_view/BuildStatisticsPortlet.java]: ws

File: src/main/java/org/jenkinsci/test/acceptance/plugins/dashboard_view/BuildStatisticsPortlet.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+    }
+  }
+
+  public BuildStatisticsPortlet(DashboardView parent, String path) {
+    super(parent, path);
+  }
+
+  /**
+   * Get the Buildstatistics table as {@link WebElement}
```
Comment (uhafner) [inline src/main/java/org/jenkinsci/test/acceptance/plugins/dashboard_view/BuildStatisticsPortlet.java]: Active: gets

File: src/main/java/org/jenkinsci/test/acceptance/plugins/dashboard_view/BuildStatisticsPortlet.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+   * @return table
+   */
+  public WebElement getTable(){
```
Comment (uhafner) [inline src/main/java/org/jenkinsci/test/acceptance/plugins/dashboard_view/BuildStatisticsPortlet.java]: Is public required or is everything accessible via getters?

File: src/main/java/org/jenkinsci/test/acceptance/plugins/dashboard_view/BuildStatisticsPortlet.java [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+    return find(By.id("statistics"));
+  }
+
+  /**
+   * Get number of builds of a specific {@link Jobtype}
+   * @param type Type of Job
+   * @return int
```
Comment (uhafner) [inline src/main/java/org/jenkinsci/test/acceptance/plugins/dashboard_view/BuildStatisticsPortlet.java]: the number of builds

File: src/main/java/org/jenkinsci/test/acceptance/plugins/dashboard_view/BuildStatisticsPortlet.java [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+   */
+  public int getNumberOfBuilds(Jobtype type){
+    return Integer.valueOf(getTable().findElement(By.xpath(".//tbody/tr["+type.row+"]/td[3]")).getText().trim());
+  }
+
+  /**
+   * Get percentage of builds of a specific {@link Jobtype}
+   * @param type Type of Job
+   * @return int
```
Comment (uhafner) [inline src/main/java/org/jenkinsci/test/acceptance/plugins/dashboard_view/BuildStatisticsPortlet.java]: the percentage

File: src/test/java/plugins/DashboardViewPluginTest.java [hunk 7]
```diff
@@ -27,4 +32,82 @@ public void configure_dashboard() {
         j.getLastBuild().shouldSucceed();
 
     }
+
+    @Test
+    public void buildStatisticsPortlet_successNr() {
+        DashboardView v = jenkins.views.create(DashboardView.class, "dashboard");
```
Comment (uhafner) [inline src/test/java/plugins/DashboardViewPluginTest.java]: Create a new 1-arg method in ViewsMixIn that uses createRandomName()

File: src/test/java/plugins/DashboardViewPluginTest.java [hunk 8]
```diff
@@ -27,4 +32,82 @@ public void configure_dashboard() {
...
+    @Test
+    public void buildStatisticsPortlet_successPercentage() {
+        DashboardView v = jenkins.views.create(DashboardView.class, "dashboard");
+        v.configure();
+        v.matchAllJobs();
+
+        BuildStatisticsPortlet stats = v.addBottomPortlet(BuildStatisticsPortlet.class);
+
+        v.save();
+
```
Comment (uhafner) [inline src/test/java/plugins/DashboardViewPluginTest.java]: In order to reduce the duplication you can extract 55-61 into a method, then the code looks like this:
```
        DashboardView v = createDashboard();
        BuildStatisticsPortlet stats = v.getBottomPortlet(BuildStatisticsPortlet.class);
```

**Review by uhafner — CHANGES_REQUESTED** [review]

File: src/main/java/org/jenkinsci/test/acceptance/plugins/dashboard_view/BuildStatisticsPortlet.java (hunk 4 above)
Comment (darenegade) [reply src/main/java/org/jenkinsci/test/acceptance/plugins/dashboard_view/BuildStatisticsPortlet.java]: I just use the public `find` method here, so i don't see a reason to block the accessibility.

File: src/main/java/org/jenkinsci/test/acceptance/plugins/dashboard_view/BuildStatisticsPortlet.java [hunk 9]
```diff
@@ -0,0 +1,58 @@
...
+        return find(By.id("statistics"));
+    }
+
+    /**
+     * Get the number of builds of a specific {@link Jobtype}
+     *
+     * @param type Type of Job
+     * @return int
+     */
+    public int getNumberOfBuilds(Jobtype type) {
```
Comment (uhafner) [inline src/main/java/org/jenkinsci/test/acceptance/plugins/dashboard_view/BuildStatisticsPortlet.java]: Maybe I'm somewhat pedantic ;-) Here is an example of a JavaDoc following my coding style: 
- Active voice
- First sentence ends with a point
- Return value is named 
- Uppercase only for actual sentences

```
/**
 * Gets the number of builds of a specific {@link Jobtype}.
 *
 * @param type type of job
 * @return number of builds
 */
```

File: src/main/java/org/jenkinsci/test/acceptance/plugins/dashboard_view/BuildStatisticsPortlet.java [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+
+/**
+ * The basic Build statistics portlet shipped with the dashboard view plugin.
+ *
+ * @author Rene Zarwel
+ */
+@Describable("Build statistics")
+public class BuildStatisticsPortlet extends AbstractDashboardViewPortlet {
+
+    public enum Jobtype {
```
Comment (uhafner) [inline src/main/java/org/jenkinsci/test/acceptance/plugins/dashboard_view/BuildStatisticsPortlet.java]: JobType CamelCase

**Review by uhafner — APPROVED** [review]

File: src/main/java/org/jenkinsci/test/acceptance/plugins/dashboard_view/BuildStatisticsPortlet.java (hunk 9 above)
Comment (darenegade) [reply src/main/java/org/jenkinsci/test/acceptance/plugins/dashboard_view/BuildStatisticsPortlet.java]: Yes, I see. I just followed the overall javadoc coding style of this project. 😄
(Just kidding - I will change that)

**Pull request merged**